src/ngamsCore/ngamsLib/pysendfile.py

```diff
@@ -190,6 +190,17 @@ def _check_sendfile_params(sock, file, offset, count):
             raise ValueError(
                 "count must be a positive integer (got %s)" % repr(count))
 
+def sendfile_send(sock, file, offset=0, count=None):
+    """sendfile_send(sock, file[, offset[, count]]) -> sent
+
+    API-compatible version of sendfile which does not use the sendfile system
+    call. The purpose of this is to allow users to disable usage of sendfile
+    (either in the case where https is used on the server and sendfile breaks,
+    or some other situation where sendfile is problematic). See the sendfile
+    function for more details about arguments.
+    """
+    return _sendfile_use_send(sock, file, offset, count)
+
```

Comment:
Unfortunately there's no easy way to detect that using sendfile will break things on the ngas level (as it skips all user-level interaction with the socket), which is why I moved it to a config file setting (so that it becomes something that can be switched on and off as needed). I suspect the best place (long term) to put the handling in a sendfile method on the cpython socket class which handles all the internals of sockets, but until that happens leaving to a config level setting is probably going to be the easiest to handle.